sql: fix handling CALL of procedures with TxnControlExpr

yuzefovich opened the pull request:
This commit addresses a couple of issues of how we handle procedures with transaction control statements:
- we forgot to plumb output columns and props to the "resumeProc" memo
- we need to ignore the result of executing `callNode` if the execution is paused for handling COMMIT or ROLLBACK statement (i.e. there will be a "resumeProc" memo).

All credit goes to Drew for figuring these things out.

Fixes: #119937.

Release note: None

yuzefovich committed:
sql: fix handling CALL of procedures with TxnControlExpr

This commit addresses a couple of issues of how we handle procedures
with transaction control statements:
- we forgot to plumb output columns and props to the "resumeProc" memo
- we need to ignore the result of executing `callNode` if the execution
is paused for handling COMMIT or ROLLBACK statement (i.e. there will be
a "resumeProc" memo).

All credit goes to Drew for figuring these things out.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/121162)
<!-- Reviewable:end -->

yuzefovich committed:
roachtest: update pgjdbc_blocklist

I have looked into a few failures of pgjdbc related to OUT parameter
support. For some, we're now running into a limitation of not allowing
usage of temporary tables within routine bodies, so we update the
blocklist accordingly.

I couldn't quite figure out why for a couple of tests pgjdbc says that
we return a different error code (we do return the correct code both for
"vanilla" invocations and prepared statements that I tried). I don't
think it's worth spending more time on this for now.

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks!

> Reviewed 20 of 20 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121162#-:-NuBNQFU6wfnjlBx_GcZ:b-g2a5fb)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
Thanks for all the help on this one!

bors r+
